### ISSUE TITLE
Unresponsive UI when using touch input [3.1.0]

### ISSUE THREAD

**Issue by st33ly** [MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-#875]
Hello,

Really loving this app as it uses way less resources on my Surface Pro 3.

I have an issue though that I haven't seen anyone else report so here goes...

When using the touch screen on my surface to select various options (e.g. 'My Library', 'New Releases', play album button, etc) more often than not the app does not respond. The input is being recognised as the menu item highlights but it takes a few taps before it works.

I know you probably didn't have touch in mind when developing this but I see no reason why this shouldn't work as the regular website works just fine. Interestingly scrolling is ok and I'm not seeing the same issue with the desktop settings popup.

Here's a video of me replicating the issue. Not the best quality but you can see where I'm tapping multiple times to access listen now - https://youtu.be/uFPm5lyiHKg

Device: Surface Pro 3 i5/4gb
OS: Windows 10
Version: 3.1.0

Thanks in advance, If this gets fixed I will donate!

**Comment by MarshallOfSound** [comment]
Does this issue occur in chrome?

I think this is just because of how touch screens, they don't have a concept of "hover" so tapping once triggers the hover event.

**Comment by st33ly** [comment]
Thanks for getting back to me so quickly!

Chrome behaves fine, I was using it briefly before I discovered your app. As I don't think the direct screen capture showed it off so well, I've recorded a video which shows a comparison between chrome and GPMDP.

Hope this helps, let me know if you need any more info

https://youtu.be/Yeh6WmNNjtE

Cheers

**Comment by MarshallOfSound** [comment]
@st33ly Can you please confirm this issue is present on `3.2.1`

**Comment by MarshallOfSound** [comment]
@st33ly Can you please confirm this is present on `3.2.4`

**Comment by st33ly** [comment]
Hi Sam,

Sorry it’s taken a while to get back to you. Prepare for a lengthy post...

I've reinstalled the latest version of GPMDP on my surface to check and I can report that there has been some improvement but overall it’s still not great on touchscreen.


Elements that work fine:

Media controls (play/pause next track, etc)
Thumbs up/down
Playlist button
Album art button
Desktop settings popup window

Track seek bar and volume sliders - the grab point on these seem to be above the actual UI element. I have to aim almost a centimetre above the orange dot to drag it successfully but otherwise works as expected. In chrome this works fine.


Elements that could be improved:

The back button doesn't always work when tapped. (On a side note, I love that you have included this but it’s not in an obvious place. I feel it would be better situated at the top left of the main window)

The burger menu button at the top right has improved but it seems like my thumb works better than fingers. It might be that I'm not hitting the sweet spot exactly but it’s hard to tell. It seems fine on chrome. Might be a long shot but if it’s possible to increase the touchable area of this button I think that might help?

The animation for the popout menu is a little sluggish on my surface pro 3 but is fine on my PC. This is possibly due to the graphics hardware of the surface but thought I’d let you know in case you can improve it?

The sidebar itself is still the same as before in that when I tap the options it will highlight but won’t actually switch over to that section. Scrolling works fine here.

Browse stations list works ok although there is slight graphical bug when scrolling where you can see a box outline. (Almost like when selecting icons on windows desktop)


Bad stuff:

I put this in a different section as I know it’s something that Google has changed for the worse which affects both chrome and GPMDP. In album and track lists, dragging my finger to scroll about now selects the album image and drags that instead. This is the same on the playlists window. I am constantly mis-scrolling or rearranging tracks by accident. I have to use the side of the window to scroll around instead.

There should really be a delay here to prevent it from happening like the android app but I guess it’s out with your control. It kinda sucks and is really bothering me. I’m guessing this behaviour changed when they revamped the logo.


So in conclusion, I can see you have been working hard on this app and I have been using it on my PC with no issue. Maybe the touchscreen support will get better in the future but I guess it depends a lot on how Google is designing the interface.

Thanks for looking into this for me. If you need any more info or testing just let me know :)

**Comment by SheepSix** [comment]
Hi Sam, I know this post is about a month old now but I found it by searching the issue.  I have the exact same issue with my Surface Pro 3 when using only touch.  Your app works perfectly when I use a mouse or the touchpad on the keyboard but when trying to just use touch on the screen itself I encounter the exact same issues as the poster above.  Your app is awesome so it's not like I'm going to stop using it, but I just wanted to confirm what the above poster is experiencing.

Cheers,

**Comment by Bohne13** [comment]
Is this bug still existing in v.3.2.5?

**Comment by MarshallOfSound** [comment]
@Bohne13 I have no idea 😆 

No way to test it 😢

**Comment by st33ly** [comment]
Hello,

So I've recently tested v3.2.5 and I'm pleased to say that the scrolling issue I mentioned above in my last post is gone which makes it a bit more usable.

Other than this, many of the same issues are still in this release:

- tracking and volume slider grab points are above the actual UI element
- back button, side menu and track play buttons dont always react to touch
- slight graphical issues when touching list items where it selects UI elements kind of like click dragging with a mouse.

all of these issues are not apparent when using GPM in chrome.

Of course the base GPM itself still has some issues with touch
- lack of ability to rearrange playlists (which was previously happening by accident)
- no back button (which GPMDP includes!)
- sluggish performance on Surface Pro 3 (Intel HD gfx)

I've been trying out players to see what works best. I used to use OutColdPlayer which was excellent but this was discontinued. The GMusic app works ok but has some bad UI decisions. So far I'm torn between using that and GPMDP with a mouse.

Its a shame that Google arent embracing desktop apps for their music service. Almost makes me want to switch over to Spotify.... :/

I'll continue to keep an eye on new releases and hopefully one of these days things will improve!

Cheers

**Comment by greatwroth** [comment]
I am using teamviewer with touch interaction enabled on my Nexus 9 to test this on Windows 10 desktop. (this nearly perfectly emulates a lag free touchscreen when used over LAN so might be worth using for testing if dev has the hardware) I can confirm several of these issues in 3.2.5.

•grab points for volume and seek bar are slightly above bars
•scrolling seems to work fine
•back button does not always work (will work eventually if tapped repeatedly)
•side menu opens and closes fine but entries do not always work (same as back button)
•double tapping to play song from main screen or jump to song in playlist does not work (this does work in chrome)

**Comment by MarshallOfSound** [comment]
@st33ly @greatwroth Had a brainwave today, can you guys please try enabling "native window borders" in the Desktop Settings, I think I know why the touch points are offset

**Comment by greatwroth** [comment]
That seems to correct everything I mentioned except the double tapping issue.

**Comment by MarshallOfSound** [comment]
@greatwroth Double tapping will be interpretted as a zoom command so I am not too worried about that.  I will raise this as an issue with electron, finally something I can explain that makes sense to me 😆

**Comment by MarshallOfSound** [comment]
electron/electron#6308

**Comment by MarshallOfSound** [comment]
@greatwroth Can you also try the latest windows build.  This bug might have been fixed upstream in Chromium

https://ci.appveyor.com/api/buildjobs/i36idv8yprdrd07l/artifacts/dist/installers/win32/Google%20Play%20Music%20Desktop%20PlayerSetup.exe

**Comment by greatwroth** [comment]
That did it. With system window borders disabled the grab points are correct, the side menu items and back button work properly, and even double tapping is working.

**Comment by MarshallOfSound** [comment]
@st33ly @greatwroth Great news, so it looks like a whole load of touch screen issues were fixed in Chromium 51.XXX  which means we will get these fixes in `3.3.0` when I release 😁

**Comment by st33ly** [comment]
@MarshallOfSound I came back from holiday to find this thread updated :)

Using system window borders has fixed most of the UI issues for me also! Looking forward to 3.3.0! :smile: :smile: :smile:
